lib/ii.c

```diff
@@ -141,6 +141,28 @@ uint8_t ii_leader_enqueue( uint8_t address
     return 0;
 }
 
+uint8_t ii_leader_enqueue_bytes( uint8_t  address
+                               , uint8_t* data
+                               , uint8_t  tx_len
+                               , uint8_t  rx_len
+                               )
+{
+    int ix = queue_enqueue( l_qix );
+    if( ix < 0 ){ printf("queue full\n"); return 1; }
+
+    ii_q_t* q = &l_iq[ix];
+
+    if( tx_len > II_MAX_BROADCAST_LEN ){ tx_len = II_MAX_BROADCAST_LEN; }
+    if( rx_len > II_MAX_RECEIVE_LEN ){ rx_len = II_MAX_RECEIVE_LEN; }
+
+    q->address = address;
+    q->query_length = rx_len;
+    q->length = tx_len;
+    memcpy( q->data, data, tx_len );
+    ii_pickle( &q->address, q->data, &q->length );
```

Comment:
tldr, we should not call ii_pickle here.

I don't think it makes sense to call `ii_pickle`.
1) We can assume that 99% of the time people won't use `ii.raw` for known devices, and ii_pickle does nothing for unknown devices
2) On the off chance that the address *is* known, the scripter should be able to control the exact bytes that are sent over the bus, rather than potentially have them jumbled.

Thinking more about use-cases, I'm guessing one might use `ii.raw` if they are communicating to a device whose address conflicts with a known device. In this case, `ii_pickle` would modify the data based on the known device, which is likely unrelated to the data expected by the conflicting device.

Reply:
> On the off chance that the address is known, the scripter should be able to control the exact bytes that are sent over the bus, rather than potentially have them jumbled.

This seemed important enough of a use case to me that I went ahead and added the `ii_q` member + global variable for disabling pickling. If I'm dropping down to raw bytes I definitely don't want the message changing! https://github.com/monome/crow/pull/367/commits/c3bea6abbca5aa76f3a401ef2d9b9fa29ec879af